### PR TITLE
Code sanity

### PR DESCRIPTION
**Pull request by priyal-unbxd**

### CODE REVIEW

File: src/modules/searchResults/renderNoResults.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+function renderNoResults(query) {
```
Comment (akshayunbxd) [inline src/modules/searchResults/renderNoResults.js:1]: query can be initialized

File: src/core/configSchema.js [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+        },
+        htmlAttributes: {
+            datatype: "object"
+        }
+    }
+}
+
+
+export const productsSchema = {
+    productType: {
```
[the remark this answers is not in the document]
Comment (priyal-unbxd) [reply src/core/configSchema.js:229]: It is required , updated it

File: src/core/configSchema.js [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+        },
+        defaultFilters: {
+            datatype: "object"
+        },
+        searchEndPoint: {
+            datatype: "string"
+        },
+        browseQueryParam: {
+            datatype: "string",
+            allowedOptions: ["p-id", "p"]
```
[the remark this answers is not in the document]
Comment (priyal-unbxd) [reply src/core/configSchema.js:592]: removed allowedOptions

File: src/modules/searchResults/renderNoResults.js (hunk 1 above)
Comment (priyal-unbxd) [reply src/modules/searchResults/renderNoResults.js:1]: Initialized it to an empty string

File: src/core/validateConfigs.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            }
+
+            //Checking if the datatype in user input config is not equal to that in metadata.
+            if (typeof userConfig[key] !== datatype || (datatype === "object" && userConfig[key] === null)) {
+                if (datatype === "element" || datatype === "array") {
+                    if (datatype === "element") {
+                        //The config is required , but the user passed either null/undefined , or the element is not present on the DOM.
+                        if (required && (!userConfig[key] || !isNode(userConfig[key]) || !isElement(userConfig[key]))) {
+                            if (userConfig[key] === window) {
+                                return null
```
[the remark this answers is not in the document]
Comment (priyal-unbxd) [reply src/core/validateConfigs.js:53]: We need to skip iteration if it is a window object , which is the case for infiniteScrollTriggerEl.

File: src/core/validateConfigs.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    facetsSchema,
+    spellCheckSchema,
+    pageSizeSchema,
+    productsSchema,
+    productViewSchema,
+    breadcrumbSchema,
+    others
+} from "./configSchema";
+
+function validateConfigs() {
```
[the remark this answers is not in the document]
Comment (priyal-unbxd) [reply src/core/validateConfigs.js:18]: using arrow function for validate configs is losing "this" context

**Pull request merged**